Source/swiftlint/Extensions/Configuration+CommandLine.swift

```diff
@@ -61,7 +61,33 @@ extension Configuration {
             let errorMessage = "No lintable files found at paths: '\(visitor.paths.joined(separator: ", "))'"
             return .failure(.usageError(description: errorMessage))
         }
-        return .success(Dictionary(grouping: files, by: configuration(for:)))
+
+        var groupedFiles = [Configuration: [File]]()
+        for file in files {
+            // Files whose configuration specifies they should be excluded will be skipped
+            let fileConfiguration = configuration(for: file)
+            let excludedPaths = fileConfiguration.excluded
+                .compactMap { fileConfiguration.rootPath?.bridge().appendingPathComponent($0) }
+
+            var shouldSkip = false
```

Comment:
Can't the entire for loop be prevented with a much more readable `contains` or `reduce` here?

I'm thinking something like:

```swift
let shouldSkip: Bool = excludedPaths.contains { file.path?.bridge().pathComponents.starts(with: $0.bridge().pathComponents) }
```